### ISSUE TITLE
Drop collection timeout if too many partitions created

### ISSUE THREAD

**Issue by del-zhenwu** [milvus-io/milvus#3319]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
[2020-08-19 10:34:23-ERROR-milvus.client.grpc_handler]: Error: <BaseException: (code=1, message=The number of partitions exceeds the upper limit (4096))> (grpc_handler.py:44)
PASSED                                                                                                                                                                              [100%]



------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------
[2020-08-19 10:35:23-ERROR-milvus.client.grpc_handler]: RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1597804523.942825996","description":"Error received from peer ipv6:[::1]:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
	{'API start': '2020-08-19 10:34:23.941416', 'RPC start': '2020-08-19 10:34:23.941645', 'RPC error': '2020-08-19 10:35:23.943434'} (grpc_handler.py:58)

test_partition.py::TestCreateBase::test_create_partition_limit ERROR                                                                                                                [100%]

========================================================================================= ERRORS ==========================================================================================
_____________________________________________________________ ERROR at teardown of TestCreateBase.test_create_partition_limit _____________________________________________________________

tp = <class 'grpc._channel._MultiThreadedRendezvous'>, value = None, tb = None

    def reraise(tp, value, tb=None):
        try:
            if value is None:
                value = tp()
            if value.__traceback__ is not tb:
                raise value.with_traceback(tb)
>           raise value

/home/zw/.local/lib/python3.6/site-packages/six.py:703: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/zw/.local/lib/python3.6/site-packages/six.py:703: in reraise
    raise value
/home/zw/.local/lib/python3.6/site-packages/six.py:703: in reraise
    raise value
conftest.py:137: in teardown
    connect.drop_collection(collection_name, timeout=delete_timeout)
/usr/local/lib/python3.6/dist-packages/milvus/client/stub.py:35: in inner
    return func(self, *args, **kwargs)
/usr/local/lib/python3.6/dist-packages/milvus/client/stub.py:347: in drop_collection
    return handler.drop_collection(collection_name, timeout)
/usr/local/lib/python3.6/dist-packages/milvus/client/grpc_handler.py:59: in handler
    raise e
/usr/local/lib/python3.6/dist-packages/milvus/client/grpc_handler.py:42: in handler
    return func(self, *args, **kwargs)
/usr/local/lib/python3.6/dist-packages/milvus/client/grpc_handler.py:373: in drop_collection
    status = rf.result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exc...1]:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
timeout = None

    def result(self, timeout=None):
        """Returns the result of the computation or raises its exception.
    
        See grpc.Future.result for the full API contract.
        """
        with self._state.condition:
            timed_out = _common.wait(self._state.condition.wait,
                                     self._is_complete,
                                     timeout=timeout)
            if timed_out:
                raise grpc.FutureTimeoutError()
            else:
                if self._state.code is grpc.StatusCode.OK:
                    return self._state.response
                elif self._state.cancelled:
                    raise grpc.FutureCancelledError()
                else:
>                   raise self
E                   grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
E                   	status = StatusCode.DEADLINE_EXCEEDED
E                   	details = "Deadline Exceeded"
E                   	debug_error_string = "{"created":"@1597804523.942825996","description":"Error received from peer ipv6:[::1]:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Deadline Exceeded","grpc_status":4}"
E                   >

/usr/local/lib/python3.6/dist-packages/grpc/_channel.py:625: _MultiThreadedRendezvous

```

**Steps/Code to reproduce behavior**
1. create collection
2. create 4097 partitions
3. drop collection

**Expected behavior**

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation (Docker, or from source)
- Milvus version (v0.3.1, or v0.4.0)
- Milvus configuration (Settings you made in `server_config.yaml`)

**Screenshots**

**Additional context**

**Comment by del-zhenwu** [comment]
```
[2020-08-20 11:52:17,822][INFO][SERVER][DropCollection][grpcpp_sync_ser] Request [21526] DropCollection begin.
[2020-08-20 11:53:19,121][INFO][SERVER][DropCollection][grpcpp_sync_ser] Request [21526] DropCollection end.
```

**Comment by sahuang** [comment]
From my own PC setup, dropping a collection with 4096 partitions costs around 23 seconds.

If partitions are halved to 2048, drop time also halved to around 11 seconds.

Since timeout is set to 60 seconds, this may be related to hardware settings. Slow I/O may lead to slower operations such as dropping partitions.

**Comment by scsven** [comment]
fix in #3393
